ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/QueryCompactor.java

```diff
@@ -115,6 +115,10 @@ void runCompactionQueries(HiveConf conf, String tmpTableName, StorageDescriptor
       }
       for (String query : compactionQueries) {
         LOG.info("Running {} compaction via query: {}", compactionInfo.isMajorCompaction() ? "major" : "minor", query);
+        if (!compactionInfo.isMajorCompaction()) {
```

Comment:
Can we leave a comment why this settings are needed?

Reply:
Sure, added a comment.